### ISSUE TITLE
Incorrect parsing of JS code within declarations

### ISSUE THREAD

**Issue by foucdeg** [hudochenkov/postcss-styled-syntax#13]
This is my code:

```typescript
export const Circle = styled.div<{ selected: boolean }>`
  background-color: ${(props) => (props.selected ? colorPalette.purple : colorPalette.black)};
`;
```

I've activated the rule 'scale-unlimited/declaration-strict-value' from https://github.com/AndyOGo/stylelint-declaration-strict-value.

Stylelint complains: "Expected variable, function or keyword for "=>" of "background-color" (scale-unlimited/declaration-strict-value)".

I will also open an issue with the rule author, but mistaking JS code for a CSS property value seems to be a parsing issue.

Using stylelint 15.6.0, postcss-styled-syntax@0.4.0, [stylelint-declaration-strict-value](https://github.com/AndyOGo/stylelint-declaration-strict-value) 1.9.2.

**Comment by hudochenkov** [comment]
This is not a bug. It is [intended behavior](https://github.com/hudochenkov/postcss-styled-syntax/blob/ca0fb8d34f95f4eb9cbd991efc58399505c5ca2a/lib/__tests__/parse.test.js#L282-L283). In PostCSS AST we have declaration with property `background-color` and value `${(props) => (props.selected ? colorPalette.purple : colorPalette.black)}`.

Since PostCSS could parse many different syntaxes, plugins should make checks on syntax to avoid false issues. E. g. Stylelint has whole bunch of `isStandard*` [utils](https://github.com/stylelint/stylelint/tree/main/lib/utils) to check if rule can work with parsed value.

**Comment by foucdeg** [comment]
Ok, thanks, so this is a plugin issue if I understand correctly?

**Comment by foucdeg** [comment]
Yep, plugin doesn't support CSS-in-JS.
